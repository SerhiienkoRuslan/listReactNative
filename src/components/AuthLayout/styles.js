import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#003f5c',
    alignItems: 'center',
    justifyContent: 'center'
  },
  logo: {
    fontWeight: 'bold',
    fontSize: 50,
    color: '#fb5b5a',
    marginBottom: 40
  },
  forgot: {
    color: 'white',
    fontSize: 11
  },
  loginBtn: {
    width: '80%',
    backgroundColor: '#fb5b5a',
    borderRadius: 25,
    height: 50,
    alignItems: 'center',
    justifyContent: 'center',
    marginTop: 40,
    marginBottom: 10
  },
  loginText: {
    color: 'white'
  }
});
