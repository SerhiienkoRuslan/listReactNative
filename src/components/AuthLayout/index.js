import React from 'react';
import { Text, TouchableOpacity, View } from 'react-native';

import routesName from 'constants/routesName';
import CustomButton from 'components/CustomButton';

import styles from './styles';

const AuthLayout = ({ children, isLogin, handleSubmit, navigation }) => {
  const handleRedirect = () =>
    navigation.navigate(
      isLogin ? routesName.REGISTRATION_SCREEN : routesName.LOGIN_SCREEN
    );

  return (
    <View style={styles.container}>
      <Text style={styles.logo}>SoWhat?</Text>

      {children}

      {isLogin && (
        <TouchableOpacity>
          <Text style={styles.forgot}>Forgot Password?</Text>
        </TouchableOpacity>
      )}

      <CustomButton
        onPress={handleSubmit}
        text={isLogin ? 'Login' : 'Sign Up'}
        customStyles={{ width: '80%', marginTop: 10, marginBottom: 10 }}
      />

      <TouchableOpacity onPress={handleRedirect}>
        <Text style={styles.loginText}>{!isLogin ? 'Login' : 'Sign Up'}</Text>
      </TouchableOpacity>
    </View>
  );
};

AuthLayout.defaultProps = {
  handleSubmit: () => {},
  isLogin: false
};

export default AuthLayout;
