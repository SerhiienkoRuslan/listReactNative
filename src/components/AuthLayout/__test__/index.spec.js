import AuthLayout from '../index';

const createTestProps = (props) => ({
  navigation: {
    navigate: jest.fn()
  },
  handleSubmit: jest.fn(),
  ...props
});

describe('AuthLayout component', () => {
  let wrapper;
  let props;
  beforeEach(() => {
    props = createTestProps({});
    wrapper = shallow(<AuthLayout {...props} />);
  });

  it('should render AuthLayout component with props', () => {
    const { getByText } = render(
      <AuthLayout
        isLogin={true}
        navigation={navigation}
        handleSubmit={handleSubmit}
      />
    );
    let buttonText = getByText('Sign Up');
    fireEvent(buttonText.parent, 'press'); //parent to get to TouchableOpacity
    alertText = getByText('Not Allowed');
    expect(alertText).toBeDefined();
  });

  it('should render AuthLayout component without props', () => {
    const component = shallow(
      <AuthLayout navigation={navigation} handleSubmit={handleSubmit} />
    );
    expect(component).toMatchSnapshot();
  });
});
