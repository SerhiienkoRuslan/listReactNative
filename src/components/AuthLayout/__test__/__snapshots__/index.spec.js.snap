// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthLayout component should render AuthLayout component with props 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#003f5c",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#fb5b5a",
        "fontSize": 50,
        "fontWeight": "bold",
        "marginBottom": 40,
      }
    }
  >
    SoWhat?
  </Text>
  <ForwardRef>
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 11,
        }
      }
    >
      Forgot Password?
    </Text>
  </ForwardRef>
  <CustomButton
    customStyles={
      Object {
        "marginBottom": 10,
        "marginTop": 10,
        "width": "80%",
      }
    }
    onPress={[MockFunction]}
    text="Login"
  />
  <ForwardRef
    onPress={[Function]}
  >
    <Text
      style={
        Object {
          "color": "white",
        }
      }
    >
      Sign Up
    </Text>
  </ForwardRef>
</View>
`;

exports[`AuthLayout component should render AuthLayout component without props 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#003f5c",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#fb5b5a",
        "fontSize": 50,
        "fontWeight": "bold",
        "marginBottom": 40,
      }
    }
  >
    SoWhat?
  </Text>
  <CustomButton
    customStyles={
      Object {
        "marginBottom": 10,
        "marginTop": 10,
        "width": "80%",
      }
    }
    onPress={[MockFunction]}
    text="Sign Up"
  />
  <ForwardRef
    onPress={[Function]}
  >
    <Text
      style={
        Object {
          "color": "white",
        }
      }
    >
      Login
    </Text>
  </ForwardRef>
</View>
`;
